File: appMain/src/main/java/com/cartlc/tracker/act/ListEntryAdapter.java
package com.cartlc.tracker.act;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.cartlc.tracker.R;
import com.cartlc.tracker.data.DataEntry;
import com.cartlc.tracker.data.DataProjectAddressCombo;
import com.cartlc.tracker.etc.PrefHelper;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by dug on 5/12/17.
 */

public class ListEntryAdapter extends RecyclerView.Adapter<ListEntryAdapter.CustomViewHolder> {

    protected class CustomViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.truck_value) TextView truckValue;
        @BindView(R.id.status)      TextView status;
        @BindView(R.id.notes)       TextView notes;
        @BindView(R.id.equipments)  TextView equipments;
        @BindView(R.id.edit)        TextView edit;

        public CustomViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }

    public interface OnItemSelectedListener {
        void onEdit(DataEntry entry);
    }

    final Context                mContext;
    final OnItemSelectedListener mListener;
    final LayoutInflater         mLayoutInflater;
    List<DataEntry> mItems;

    public ListEntryAdapter(Context context, OnItemSelectedListener listener) {
        mContext = context;
        mListener = listener;
        mLayoutInflater = LayoutInflater.from(mContext);
    }

    @Override
    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = mLayoutInflater.inflate(R.layout.entry_item_full, parent, false);
        return new CustomViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final CustomViewHolder holder, final int position) {
        final DataEntry item = mItems.get(position);
        holder.truckValue.setText(item.getTruckLine(mContext));
        holder.status.setText(item.getStatus(mContext));
        holder.notes.setText(item.getNotesLine());
        holder.equipments.setText(item.getEquipmentLine(mContext));
        holder.edit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mListener.onEdit(item);
            }
        });
        if (TextUtils.isEmpty(holder.notes.getText().toString().trim())) {
            holder.notes.setVisibility(View.GONE);
        } else {
            holder.notes.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    public void onDataChanged() {
        DataProjectAddressCombo combo = PrefHelper.getInstance().getCurrentProjectGroup();
        if (combo == null) {
            mItems = new ArrayList();
        } else {
            mItems = combo.getEntries();
        }
    }
}

File: appMain/src/main/java/com/cartlc/tracker/ui/list/ListEntryAdapter.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.ui.list

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.cartlc.tracker.R
import com.cartlc.tracker.fresh.model.CarRepository
import com.cartlc.tracker.fresh.model.core.data.DataEntry
import com.cartlc.tracker.fresh.model.pref.PrefHelper
import com.cartlc.tracker.ui.app.TBApplication
import kotlinx.android.synthetic.main.entry_item_full.view.*

/**
 * Created by dug on 5/12/17.
 */

class ListEntryAdapter(
        private val mContext: Context,
        private val mListener: OnItemSelectedListener
) : RecyclerView.Adapter<ListEntryAdapter.CustomViewHolder>() {
    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(mContext)
    private var mItems: List<DataEntry> = emptyList()

    private val app: TBApplication
        get() = mContext.applicationContext as TBApplication
    val repo: CarRepository
        get() = app.repo
    private val prefHelper: PrefHelper
        get() = repo.prefHelper

    inner class CustomViewHolder(val view: View) : RecyclerView.ViewHolder(view) {

            fun bind(item: DataEntry) {
                with(view) {
                    truck_value!!.text = item.getTruckLine(mContext)
                    status!!.text = item.getStatus(mContext)
                    notes!!.text = item.notesLine
                    equipments!!.text = item.getEquipmentLine(mContext)
                    edit!!.setOnClickListener { mListener.onEdit(item) }
                    if (TextUtils.isEmpty(notes!!.text.toString().trim { it <= ' ' })) {
                        notes!!.visibility = View.GONE
                    } else {
                        notes!!.visibility = View.VISIBLE
                    }
                }
            }
    }

    interface OnItemSelectedListener {
        fun onEdit(entry: DataEntry)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
        val view = mLayoutInflater.inflate(R.layout.entry_item_full, parent, false)
        return CustomViewHolder(view)
    }

    override fun onBindViewHolder(holder: CustomViewHolder, position: Int) {
        holder.bind(mItems[position])
    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    fun onDataChanged() {
        val combo = prefHelper.currentProjectGroup
        if (combo == null) {
            mItems = emptyList()
        } else {
            mItems = combo.entries
        }
    }
}
